Kitodo-DataAccess/src/test/java/org/kitodo/dataaccess/NodeReferenceTest.java

```diff
@@ -0,0 +1,134 @@
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.dataaccess;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+import org.kitodo.dataaccess.NodeReference;
+import org.kitodo.dataaccess.Storage;
+
+public class NodeReferenceTest {
+
+    @Test
+    public void testEqualsIsFalseForDifferentObjects() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            NodeReference one = storage.createNodeReference("http://names.example/foo");
+            NodeReference other = storage.createNodeReference("http://names.example/bar");
+
+            assertThat(one, is(not(equalTo(other))));
+        }
+    }
+
+    @Test
+    public void testEqualsObject() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            NodeReference one = storage.createNodeReference("http://names.example/foo");
+            NodeReference another = storage.createNodeReference("http://names.example/foo");
+
+            assertThat(one, is(equalTo(another)));
+        }
+    }
+
+    @Test
+    public void testGetIdentifier() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            assertEquals("http://names.example/foo",
+                storage.createNodeReference("http://names.example/foo").getIdentifier());
+        }
+    }
+
+    @Test
+    public void testHashCodeDiffersForTwoDifferentInstances() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            NodeReference one = storage.createNodeReference("http://names.example/foo");
+            NodeReference other = storage.createNodeReference("http://names.example/bar");
+
+            assertThat(one.hashCode() == other.hashCode(), is(false));
+        }
+    }
+
+    @Test
+    public void testHashCodeIsEqualForTwoEqualInstances() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            NodeReference one = storage.createNodeReference("http://names.example/foo");
+            NodeReference other = storage.createNodeReference("http://names.example/foo");
+
+            assertThat(one.hashCode() == other.hashCode(), is(true));
+        }
+    }
+
+    @Test
+    public void testNodeReferenceCanBeCreatedFromHttpUrl() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            storage.createNodeReference("http://names.example/foo#bar");
+        }
+    }
+
+    @Test
+    public void testNodeReferenceCanBeCreatedFromNonHttpUrl() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            storage.createNodeReference("urn:example:kitodo-123456789-xyz");
```

Comment:
What you are testing here? Did not look like a test as no comparison or assertion is done.

Reply:
No assertion necessary. Test is to prove that the method does not throw an exception, what it shall do only if the string in the constant would not be an URI. (Tests that URNs pass the URI validation correctly, not giving false positives.)